### PR TITLE
py_wrap: implement nested class definitions

### LINKED ISSUE #4670: Pyosys build breaks: Const has no member named 'operator*'
### Version

main 81bbde62

### On which OS did this happen?

Linux

### Reproduction Steps

Build yosys with ENABLE_PYOSYS=1

My almost-full Makefile.conf:
```
CONFIG := clang
ENABLE_VERIFIC := 1
ENABLE_CCACHE := 1
ENABLE_LTO=1
ENABLE_PYOSYS=1
```

### Expected Behavior

Happy build

### Actual Behavior

Sad build

### PR DESCRIPTION
**Pull request by widlarizer**
Fixes #4670 by probably implementing nested class declaration. I don't really know what the wrapper generator does and I don't want to find out more than I already did. Also the DIY C++ parser doesn't like trailing commas in enum declarations so I removed it from `RTLIL::ConstFlags`.

### CODE REVIEW

**Comment by povik** [comment]
Is there a diff in the generator's output against from it produced the last time it worked?

File: misc/py_wrap_generator.py [hunk 1]
```diff
@@ -2034,28 +2034,31 @@ def parse_header(source):
 				debug("\t  " + struct_name + " is derived from " + base_class_name,2)
 			base_class = class_by_name(base_class_name)
 
-			class_ = (class_by_name(struct_name), ugly_line.count("{"))#calc_ident(line))
+			c = (class_by_name(struct_name), ugly_line.count("{"))#calc_ident(line))
 			debug(f"switch to {struct_name} in namespace {namespaces}", 2)
 			if struct_name in classnames:
-				class_[0].namespace = complete_namespace
-				class_[0].base_class = base_class
+				c[0].namespace = complete_namespace
```
[the remark this answers is not in the document]
Comment (widlarizer) [reply misc/py_wrap_generator.py:2040]: It's been like that before. Here's how we're getting away with it:

```
>>> c = (1, 2)
>>> c[0] = 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'tuple' object does not support item assignment
>>> c = ([1, 2], [3, 4])
>>> c[0].append(9)
>>> c
([1, 2, 9], [3, 4])
```

[the remark this answers is not in the document]
Comment (povik) [reply misc/py_wrap_generator.py:2040]: Makes sense

**Comment by widlarizer** [comment]
Here you go. The diff is the const additions and then temporary variable naming noise

[python_wrappersnew.txt](https://github.com/user-attachments/files/17421908/python_wrappersnew.txt)
[python_wrappersold.txt](https://github.com/user-attachments/files/17421909/python_wrappersold.txt)

**Comment by povik** [comment]
The flags property seems to have disappeared

**Comment by widlarizer** [comment]
`short` -> `short int` fixes that

**Comment by povik** [comment]
Not the change I expected to help

**Review by povik — APPROVED** [review]
The generator output looks good so I think we can merge this

**Pull request merged**